core/domain/beam_job_services.py

```diff
@@ -136,12 +136,21 @@ def get_beam_job_run_result(job_id):
     Returns:
         BeamJobRunResult. The result of the given Apache Beam job run.
     """
-    beam_job_run_result_model = (
-        beam_job_models.BeamJobRunResultModel.get(job_id, strict=False))
+    beam_job_run_result_models = beam_job_models.BeamJobRunResultModel.query(
+        beam_job_models.BeamJobRunResultModel.job_id == job_id).iter()
+
+    # Job results are inherently unordered; there's no need to sort them.
+    stdout, stderr = [], []
+    for beam_job_run_result_model in beam_job_run_result_models:
+        if beam_job_run_result_model.stdout:
+            stdout.append(beam_job_run_result_model.stdout)
+        if beam_job_run_result_model.stderr:
+            stderr.append(beam_job_run_result_model.stderr)
+
     return (
-        None if beam_job_run_result_model is None else
+        None if not stdout and not stderr else
         beam_job_domain.BeamJobRunResult(
-            beam_job_run_result_model.stdout, beam_job_run_result_model.stderr))
+            stdout='\n'.join(stdout), stderr='\n'.join(stderr)))
```

Comment:
Isn't there still a possibility here that the concatenated stdout/stderr will overflow the 1MiB limit for an NDB model instance?

Reply:
Well that's not a problem here because these are domain objects, not NDB instances.

Reply:
Yup but at some point the domain object will get stored in NDB right (BeamJobRunResultModel) -- what happens then?

Might be worth having an explicit backend test that checks the case with > 1 MiB of output.

Reply:
No, we won't be using these domain objects to store results into storage, that will be a separate type in the `jobs/` directory. This output is reserved for the Frontend exclusively.

Should I add documentation to make that clear?

Reply:
Yes, probably. And yup, I think I see now, we're storing max 1 MIB of data in each storage model and this just groups all the stuff for the benefit of the frontend?

Reply:
Yep that's exactly right. Maybe `AggregatedBeamJobRunResults`?

Reply:
Yup that sounds better, thanks!